lib/order.rb

```diff
@@ -0,0 +1,55 @@
+require_relative 'menu'
+
+class Order
+  attr_reader :menu, :cost, :items
+  DELIVERY_TIME = 1800
+
+  def initialize(menu = Menu.new)
+    @menu = menu.list
+    @items = {}
+    cost = 0
+  end
+
+  def update(items)
+    manage_order(items)
+    calculate_cost
+  end
+
+  def confirm
+    "Thank you! Your order was placed and will be delivered before #{Time.new + DELIVERY_TIME}."
+  end
+
+  private
+
+  def manage_order(items)
+    items = items.split(", ")
+    items.each { |dish|
+    reformat_order(dish)
+    }
+  end
+
+  def reformat_order(dish)
+    dishes = dish.split(" ")
+    dish = dishes[1].to_sym
+    number = dishes[0].to_i
+    check_order(dish, number)
+    save_order(dish, number)
+  end 
+
+  def check_order(dish, number)
+    fail "#{dish} is not on the menu" unless @menu.include?(dish)
+  end
+
+  def save_order(dish, number)
+    @items[dish] = number
+  end
+
+  def calculate_cost
+    @cost = 0
+    @items.each { |meal, number|
+      item_cost = @menu[meal].split("£")[1].to_i
```

Comment:
You could look at. not includign '£' in the hash, but leave it as an integer and then add the '£' when printing.